lib/expression/callbacks/standard.ex

```diff
@@ -100,6 +114,74 @@ defmodule Expression.Callbacks.Standard do
     end
   end
 
+  @doc """
+  Calculates a new datetime based on a base date, a desired day and a desired time.
+
+  If no base date is supplied, it will fallback to current time.
+  """
+  @expression_doc doc:
+                    "Shifts a datetime to the next occurrence of a given day of the week and a set time.",
+                  expression: "next(\"monday\", \"10:00\", datetime)",
+                  context: %{
+                    "datetime" => ~U[2023-02-03 20:18:03Z],
+                    "number" => %{"from_addr" => "+5521987724322"}
+                  },
+                  fake_result: Map.merge(@datetime_struct, %{day: 6, hour: 10, minute: 0})
+  @expression_doc doc:
+                    "If the day of the week is the same as the base date, the next occurrence is 7 days later.",
+                  expression: "next(\"thursday\", \"10:30\", datetime)",
+                  context: %{
+                    "datetime" => ~U[2023-02-02 20:18:03Z],
+                    "number" => %{"from_addr" => "+5521987724322"}
+                  },
+                  fake_result: Map.merge(@datetime_struct, %{day: 9, hour: 10, minute: 30})
+  @spec next(map, {:literal, String.t()}, {:literal, String.t()}, any) :: DateTime.t()
+  def next(
+        %{"number" => %{"from_addr" => from_addr}} = ctx,
```

Comment:
Maybe these no local return things that Dialyzer is complaining about relate to this?

Reply:
Also `number.from_addr` is quite Turn specific. In FLOIP land, the contact's details would be under the `%{"contact" => %{...}}`. I'd need to check what the expected key of the address of the contact is.

Reply:
Yeah, I guess it would be better to get the recipients' number. What would be the correct field for that?

Reply:
According to the the FLOIP spec it's https://floip.gitbook.io/flow-specification/context#contact `contact.phonenumber` but I don't think we set that yet in Turn.

Our options are either to introduce that field into Turn (it's currently in contact.whatsapp_id in Turn) or moving this helper to the Turn callbacks module (outside of this repo) so that we make it easier to keep track of customisations & standard deviations in a bit more centralised fashion in the Turn codebase.

Reply:
Moving this to the custom callbacks function in Turn also has the benefit of not needing to copy the phone number prefix stuff below.

Reply:
I didn't know that was even an option. But yeah, I think I'm gonna do that @smn